deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/params/DeconvolutionParamInitializer.java

```diff
@@ -0,0 +1,85 @@
+package org.deeplearning4j.nn.params;
+
+import org.deeplearning4j.nn.conf.NeuralNetConfiguration;
+import org.deeplearning4j.nn.conf.distribution.Distributions;
+import org.deeplearning4j.nn.weights.WeightInitUtil;
+import org.nd4j.linalg.api.ndarray.INDArray;
+import org.nd4j.linalg.api.rng.distribution.Distribution;
+import org.nd4j.linalg.indexing.NDArrayIndex;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public class DeconvolutionParamInitializer extends ConvolutionParamInitializer {
```

Comment:
I see DeconvolutionParamInitializer.WEIGHT_KEY (and bias) used elsewhere, but these are not defined here??

Reply:
The only difference is in the _shape_ of those weights, compare in `createWeightMatrix`:

deconv:
```
int[] weightsShape = new int[] {inputDepth, outputDepth, kernel[0], kernel[1]};
```

conv:
```
 int[] weightsShape = new int[] {outputDepth, inputDepth, kernel[0], kernel[1]};

```

Reply:
For this reason I inherit from `ConvolutionParamInitializer`, to keep all other things equal.